modules/getComponents.js

```diff
@@ -8,7 +9,13 @@ function getComponentsForRoute(nextState, route, callback) {
 
   const getComponent = route.getComponent || route.getComponents
   if (getComponent) {
-    getComponent.call(route, nextState, callback)
+    const componentReturn = getComponent.call(route, nextState, callback)
+    if (isPromise(componentReturn))
+      componentReturn
+        // Try module.default first in case of System.import and Babel 6
+        .then(component => component.default || component)
```

Comment:
I don't think it's Babel 6 – I think that's just what `System.import` does, no? What happens if there are non-default exports.

I wonder if we should handle this though. Presumably users can do

```js
function asyncComponent(id) {
  return System.import(id).then(component => component.default);
}
```